test/unit/clarity_ext/dilution/test_dilutions.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+from mock import MagicMock
+from clarity_ext.dilution import DilutionScheme
+from test.unit.clarity_ext.helpers import fake_analyte
+
+
+class TestDilutionScheme(unittest.TestCase):
+    """
+    Validates that the DilutionScheme property of the context object returns the expected values
+    based on fake data from the LIMS
+    """
+
+    def test_dilution_scheme_hamilton_expected_results(self):
+        """Dilution scheme created by mocked analytes is correctly generated for Hamilton"""
+        # Setup:
+        repo = MagicMock()
+        repo.all_analytes = all_analyte_test_set1
+        dilution_scheme = DilutionScheme(repo, "Hamilton")
+
+        expected = [
+            ['art-name1', '36', 'DNA1', '14.9', '5.1', '34', 'END1'],
+            ['art-name2', '33', 'DNA2', '14.9', '5.1', '17', 'END2'],
+            ['art-name3', '50', 'DNA2', '14.9', '5.1', '44', 'END1'],
+            ['art-name4', '93', 'DNA2', '14.9', '5.1', '69', 'END2']]
+
+        # Test:
+        actual = [
+            [dilute.sample_name,
+             "{}".format(dilute.source_well_index),
+             dilute.source_plate_pos,
+             "{:.1f}".format(dilute.sample_volume),
+             "{:.1f}".format(dilute.buffer_volume),
+             "{}".format(dilute.target_well_index),
+             dilute.target_plate_pos] for dilute in dilution_scheme.dilutes
+        ]
```

Comment:
That's a valid concern. 

My idea is that the code in the private repo should generally do almost nothing, only format, reorder and so on - specifics for the configuration of the devices we have. So it should perhaps read more like configuration rather than code. Those specifics will not be likely to be reused in exactly that format (if they ever are, I think the feature should then be pushed down to clarity-ext). In this case, the code is actually partially a copy of the extension code, but the beauty is that the extension can change at will, while this test would not change, still validating that `DilutionScheme` actually is doing the correct thing.

The power in the split between the library and the particular extensions (configuration) built from it will only be clear if anyone ever uses the underlying library. Although I'm not sure anyone else will ever use it, I am sure that no-one will use it if it's not correctly broken up from the beginning (e.g. if it will contain a ton of code that's only interesting for us.)

The code that I did copy is the order of the values and the formatting. I would actually prefer that this test would not do that formatting, then it would not really be duplication. I kept it in there to make it clear that this actually tests what the old code tested - but I think it wasn't valid, so I'll remove the formatting to make this clear.